examples/recursion/Go/fibonacci-memoization.go

```diff
@@ -0,0 +1,26 @@
+package main
```

Comment:
Create a wrapper function for the recursive function here rather than having the memoization table being global (see this example in Python: https://github.com/johnfraserss/ICS4U/blob/master/examples/recursion/Python/recursive_memoization.py)